cloudevents/pydantic/event.py

```diff
@@ -0,0 +1,305 @@
+            " they all use the same algorithm to determine the value used."
+        ),
+        example="2018-04-05T17:31:00Z",
+    )
+
+    subject: typing.Optional[str] = pydantic.Field(
+        title="Event Subject",
+        description=(
+            "This describes the subject of the event in the context of the event"
+            " producer (identified by source). In publish-subscribe scenarios, a"
+            " subscriber will typically subscribe to events emitted by a source, but"
+            " the source identifier alone might not be sufficient as a qualifier for"
+            " any specific event if the source context has internal"
+            " sub-structure.\n"
+            "\n"
+            "Identifying the subject of the event in context"
+            " metadata (opposed to only in the data payload) is particularly helpful in"
+            " generic subscription filtering scenarios where middleware is unable to"
+            " interpret the data content. In the above example, the subscriber might"
+            " only be interested in blobs with names ending with '.jpg' or '.jpeg' and"
+            " the subject attribute allows for constructing a simple and efficient"
+            " string-suffix filter for that subset of events."
+        ),
+        example="123",
+    )
+    datacontenttype: typing.Optional[str] = pydantic.Field(
+        title="Event Data Content Type",
+        description=(
+            "Content type of data value. This attribute enables data to carry any type"
+            " of content, whereby format and encoding might differ from that of the"
+            " chosen event format."
+        ),
+        example="text/xml",
+    )
+    dataschema: typing.Optional[str] = pydantic.Field(
+        title="Event Data Schema",
+        description=(
+            "Identifies the schema that data adheres to. "
+            "Incompatible changes to the schema SHOULD be reflected by a different URI"
+        ),
+    )
+
+    def __init__(
+        self,
+        attributes: typing.Optional[typing.Dict[str, typing.Any]] = None,
+        data: typing.Optional[typing.Any] = None,
+        **kwargs
+    ):
+        """
+        :param attributes: A dict with CloudEvent attributes.
+            Minimally expects the attributes 'type' and 'source'. If not given the
+            attributes 'specversion', 'id' or 'time', this will create
+            those attributes with default values.
+
+            If no attribute is given the class MUST use the kwargs as the attributes.
+
+            Example Attributes:
+            {
+                "specversion": "1.0",
+                "type": "com.github.pull_request.opened",
+                "source": "https://github.com/cloudevents/spec/pull",
+                "id": "A234-1234-1234",
+                "time": "2018-04-05T17:31:00Z",
+            }
+
+        :param data: Domain-specific information about the occurrence.
+        :type data: typing.Any
```

Comment:
Let's not add `type` when we already have typings. This is a duplicate that's gonna be harder to manage.